app/scripts/controllers/app-state.js

```diff
@@ -401,6 +405,29 @@ export default class AppStateController extends EventEmitter {
     this.store.updateState({ showAccountBanner });
   }
 
+  /**
+   * Sets an object with networkName and appName
+   * or `null` if the message is meant to be cleared
+   *
+   * @param switchedNetworkDetails
+   */
+  setSwitchedNetworkDetails(switchedNetworkDetails) {
```

Comment:
Nit: Generally I'd advise against this type of method being added to a controller. We try to model _user actions_ with methods, rather than using them for key-value storage. For example, these two methods could be split into three methods: `trackAutomaticNetworkSwitch`, `clearAutomaticNetworkSwitch`, and `neverShowAutomaticNetworkSwitch`. 

This is the same guideline we use for Redux: https://redux.js.org/style-guide#model-actions-as-events-not-setters

In addition to the advantages listed there, this style of method sets clearer boundaries for the controller's area of responsibility: events happen external to the controller, and the controller should update state accordingly, keeping state consistent at all times.

Note a blocker here though, I see that setters/getters are already a norm for this controller.

I see that setters and geters are al

Reply:
Can address this in a follow-up.  As you said, that's par for the course in this controller thus far, though not ideal.  To keep the project moving ahead, will do in a follow up later.